app/views/events/show.html.erb

```diff
@@ -1,6 +1,6 @@
 <%- model_class = Event -%>
 <div class="page-header">
-  <h1><%= t '.title', :default => model_class.model_name.human.titleize %></h1>
+  <h1><%= @event.kind.humanize %> <%=t '.title', :default => model_class.model_name.human.titleize %></h1>
```

Comment:
I'll put it last, in parenthesis. So we would get "Mint Camp (Camp)" which is still kinda stupid, but at least explicit for the user.